conda/common/io.py

```diff
@@ -689,6 +693,114 @@ def _ensure_dir(self):
             os.makedirs(dirname(self.record_file))
 
 
+Reporter = Dict[str, Union[bool, int, str]]
+DetailRecord = Dict[str, Union[str, int, bool]]
+
+
+class ReporterHandlerBase(ABC):
+    """
+    Base class for all reporter handlers
+    """
+
+    @abstractmethod
+    def detail_view(self, data: DetailRecord, **kwargs) -> str: ...
```

Comment:
Can we add comments for these two methods? (`detail_view` and `string_view`) It would be good to know why they exists and how they are expected to be implemented. When authors are creating their own sub-classes based on this class they will be looking to these comments for guidance.